osu.Game/Online/API/Requests/GetScoresRequest.cs

```diff
@@ -37,17 +41,19 @@ private void onSuccess(APILegacyScores r)
                 score.Beatmap = beatmap;
         }
 
-        protected override WebRequest CreateWebRequest()
+        protected override string Target => $@"beatmaps/{beatmap.OnlineBeatmapID}/scores{createQueryParameters()}";
+
+        private string createQueryParameters()
         {
-            var req = base.CreateWebRequest();
+            StringBuilder query = new StringBuilder(@"?");
 
-            req.Timeout = 30000;
-            req.AddParameter(@"type", scope.ToString().ToLowerInvariant());
-            req.AddParameter(@"mode", ruleset.ShortName);
+            query.Append($@"type={scope.ToString().ToLowerInvariant()}&");
+            query.Append($@"mode={ruleset.ShortName}&");
 
-            return req;
-        }
+            foreach (var mod in mods)
+                query.Append($@"mods[]={mod.Acronym}&");
 
-        protected override string Target => $@"beatmaps/{beatmap.OnlineBeatmapID}/scores";
+            return query.ToString().TrimEnd('&');
```

Comment:
Super minor thing: Rather than trimming you can put the `&` in front where you iterate mods and omit it on the mode.